tools/repl.nim

```diff
@@ -154,11 +156,19 @@ proc process(ctx: var ModuleCtx, reporter: Reporter,
       echo "validation failure"
       return
 
-    var env = initVm(1024, 1024 * 1024)
+    var mem: MemoryConfig
+    if (let v = readMemConfig(module); v.isSome):
+      mem = v.unsafeGet
+    else:
+      # if reached, there's a bug in either source2il or the lowering passes
+      unreachable()
```

Comment:
```suggestion
      unreachable:
        "failed to read MemoryConfig: bug in either source2il or the lowering passes"
```

soft suggestion, just slightly nicer to debug

Reply:
I've slightly shortened the string; a pass bug is a lot less likely (due to their separate tests) than a `source2il` one.